apollo-client/src/main/java/com/ctrip/framework/apollo/internals/AbstractConfig.java

```diff
@@ -494,6 +513,46 @@ private boolean isConfigChangeListenerInterested(ConfigChangeListener configChan
     return false;
   }
 
+  /**
+   * @return new {@link ConfigChangeEvent} contains interested and changed keys
+   */
+  private ConfigChangeEvent resolve(ConfigChangeListener configChangeListener,
+      ConfigChangeEvent configChangeEvent) {
+    if (!this.m_interestedKeys.containsKey(configChangeListener) && !this.m_interestedKeyPrefixes.containsKey(configChangeListener)) {
+      return configChangeEvent;
+    }
+
+    Set<String> interestedChangedKeys = resolveInterestedChangedKeys(
+        configChangeEvent.changedKeys(),
+        m_interestedKeys.containsKey(configChangeListener) ? m_interestedKeys
+            .get(configChangeListener) : Collections.<String>emptySet(),
+        m_interestedKeyPrefixes.containsKey(configChangeListener) ? m_interestedKeyPrefixes
+            .get(configChangeListener) : Collections.<String>emptySet()
+    );
+    return new ConfigChangeEvent(configChangeEvent.getNamespace(), configChangeEvent.getChanges(),
+        interestedChangedKeys);
+  }
+
+  static Set<String> resolveInterestedChangedKeys(Set<String> changedKeys, Set<String> interestedKeys, Set<String> interestedKeyPrefixes) {
```

Comment:
personally I would prefer non-static methods since it's easier to be mocked and tested.